pkg/rbac/cache.go

```diff
@@ -15,6 +15,7 @@ type Cache struct {
 	tokenReviews     map[string]*tokenReviewCache
 	tokenReviewsLock sync.Mutex
 	shared           clusterScopedResources
+	users            namespacedResources
```

Comment:
1. We'll need to save more data for each user, so instead of `namespacedResources` let's use a more general struct like `userData`.  Then `namespacedResources` is inside `userData`
2. Let's make this a `map[string] *userData`.  The map key will be the user UID that we get from the TokenReview.